pkg/api/convertertoagentpoolonlyapi.go

```diff
@@ -510,3 +516,15 @@ func convertV20180331AgentPoolOnlyAddonProfiles(obj map[string]v20180331.AddonPr
 	}
 	return api
 }
+
+func convertV20180331AgentPoolOnlyAADProfile(obj *v20180331.AADProfile) *AADProfile {
+	aadProfile := &AADProfile{
```

Comment:
We can just return this struct directly:
```go
return &AADProfile {
...
```

Reply:
Updated. Pushed new changes PTAL.